### PR TITLE
Update core global styles panel events for new UI

### PR DESCRIPTION
**Pull request by Addison-Stavlo**
#### Changes proposed in this Pull Request

Updates global styles panel events to fit with the newly updated interface.  Previously, we used 1 event to track both open/close of the panel as well as the selected tab context.  Now that tabs have been removed it makes sense to decouple this into two events.  1 for toggling the panel open/closed states, and another for selection of a menu context (typography, colors, blocks, etc.).

follow-up TODOs - after 11.6 lands on production, we merge this PR, and release the new `wpcom-block-editor` build:
* update tracks to remove the `wpcom_block_editor_global_styles_tab_selected` and add our two new replacement events `wpcom_block_editor_global_styles_panel_toggle` and `wpcom_block_editor_global_styles_menu_selected`.
* open another PR to test and release the updated e2e tests that are added with a `.skip` here. 

#### Testing instructions

* Build the `wpcom-block-editor` app and sync it to your sandbox.  
* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Visit the site editor on a site running gutenberg edge (or Gutenberg v11.6.1-rc.1 or greater).
* Open the global styles panel and verify the `wpcom_block_editor_global_styles_panel_toggle` event fires with the property `open` as `true`.
* Close the global styles panel and verify this same event fires with the `open` property as `false`.
* Open the global styles panel, then open the block inspector panel, verify the above event also fires with the `open` property as `false`.
* Open the global styles panel and select one of the menu options (Typography, Colors, or Blocks).  Verify that for this action the `wpcom_block_editor_global_styles_menu_selected` event fires with the `menu` property corresponding to the menu selected ('typography', 'colors', or 'blocks').

Related to #

### CODE REVIEW

File: apps/wpcom-block-editor/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
 		"bundle": "BROWSERSLIST_ENV=wpcom calypso-build --env WP",
 		"build:dev": "yarn run bundle",
 		"build:prod": "NODE_ENV=production yarn run bundle",
-		"build": "yarn clean && npm-run-all --parallel 'build:*'",
+		"build": "yarn clean && npm-run-all --parallel \"build:* {@}\" --",
```
Comment (Addison-Stavlo) [inline apps/wpcom-block-editor/package.json]: Note this change is temporary in this PR as the issue it resolves is being fixed elsewhere.  Updating this was required to run the dev build command in the meantime.

File: test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js [hunk 2]
```diff
@@ -415,98 +428,74 @@ describe( `[${ host }] Calypso Gutenberg Site Editor Tracking: (${ screenSize })
...
-			} );
-
-			it( 'when Global Styles sidebar is closed', async function () {
-				const editor = await SiteEditorComponent.Expect( this.driver );
-
-				// Note the sidebar is already open here because of the previous test.
-				await editor.toggleGlobalStyles();
+		// Temporarily skip these tests until we can deploy wpcom-block-editor changes from
+		// https://github.com/Automattic/wp-calypso/pull/56551
+		describe.skip( 'Global styles panel events', function () {
```
Comment (Addison-Stavlo) [inline test/e2e/specs/specs-wpcom/wp-calypso-gutenberg-site-editor-tracking-spec.js:433]: We will have to wait to test these updated tests in a subsequent PR to remove the `.skip` after the `wpcom-block-editor` changes are released.  Currently I cannot get the site editor to load in the e2e test environment when sandboxing widgets, similarly we would not want the tests running and failing after this PR is merged yet before the `wpcom-block-editor` build is released.

File: apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-global-styles-menu-selected.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+		'.edit-site-global-styles-sidebar .edit-site-global-styles-preview'
+	);
+
+	if ( ! isAtTopLevel ) {
+		return;
+	}
+
+	const buttonText = path.find( ( node ) => node.nodeName === 'BUTTON' )?.innerText;
+
+	if ( buttonText === __( 'Typography' ) ) {
```
Comment (zaguiini) [inline apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-global-styles-menu-selected.js:15]: I wonder if is there a way to get the original string out of the translated value. Like, how can I get the `Typography` key back using the `Tipografia` translated value?

If that's possible, we could transform those `if`s into a single call passing the translation key as the `menu` property's value, like:

```ts
tracksRecordEvent( 'wpcom_block_editor_global_styles_menu_selected', {
    menu: getTranslationKey(buttonText),
} );
```

Which would make the code :chefs-kiss:

Don't get me wrong, I actually think your current code is great and an explicit solution -- I'm just thinking about how could we scale it better if it ever grows in the future without needing to change the code.

Comment (Addison-Stavlo) [reply apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-global-styles-menu-selected.js:15]: Thats a great idea and would definitely be neat!  👩‍🍳 💋  code. 😆 

I don't see anything like that in the i18n package, but that could be a really cool addition.  I wonder how much use that would get beyond a less common situation like this.

**Review by zaguiini — APPROVED** [review]
Works well!

I updated your PR's description to match the event name on the follow-up part -- `styled` instead of `styles`.

**Pull request merged**